src/framework/components/element/system.js

```diff
@@ -260,6 +264,7 @@ Object.assign(pc, function () {
                 sprite: source.sprite,
                 spriteFrame: source.spriteFrame,
                 pixelsPerUnit: source.pixelsPerUnit,
+                key: source.key,
```

Comment:
If 'text' exists in the source and it's the translated text then key will exist too. If 'text' exist in the source because the user set it directly then key will be null. So when you clone it it should either use the key with a higher priority if it exists or use the text.

Reply:
Even if both key and text are set as a correct pair. When this calls initializeComponent setting "text" will clear the key to null.